spec/services/login_spec.rb

```diff
@@ -4,33 +4,63 @@
   include PermittedDomainHelper
 
   let(:service) { described_class.new(session, person) }
-
-  let(:session) { {} }
   let(:person) { create(:person) }
+  let(:session) { {} }
 
   describe "#login" do
     subject(:login) { service.login }
 
-    let(:current_time) { Time.zone.now }
+    context "when an Moj user logs in" do
+      let(:current_time) { Time.zone.now }
+
+      it "increments login count" do
+        expect { login }.to change(person, :login_count).by(1)
+      end
+
+      it "stores the current time of login" do
+        Timecop.freeze(current_time) do
+          expect { login }.to change(person, :last_login_at)
+          expect(person.last_login_at.change(usec: 0)).to eq(current_time.change(usec: 0))
+        end
+      end
+
+      it "the session key changes" do
+        expect { login }.to change { session[Login::SESSION_KEY] }.from(nil).to(person.id)
+      end
 
-    it "increments login count" do
-      expect { login }.to change(person, :login_count).by(1)
+      it "the key type changes" do
+        expect { login }.to change { session[Login::TYPE_KEY] }.from(nil).to("person")
+      end
     end
 
-    it "stores the current time of login" do
-      Timecop.freeze(current_time) do
-        expect { login }.to change(person, :last_login_at)
-        expect(person.last_login_at.change(usec: 0)).to eq(current_time.change(usec: 0))
+    context "when an External user logs in" do
+      let(:person) { create(:external_user) }
+
+      it "the session key changes" do
```

Comment:
It's not always possible, but if the test name can be written so it makes sense as a sentence that is ideal.
e.g. `it "changes the session key" do`

There are a few tests in the file that could be updated like that.

Reply:
Ahh, I was ignoring the 'it' part.  The test descriptions in this file have now been updated.